File: 0969-pancake-sorting/0969-pancake-sorting.py
class Solution:
    def __init__(self):
        self.result = []

    def pancakeSort(self, arr: List[int]) -> List[int]:
        N = len(arr)

        while N > 1:
            max_index = arr.index(max(arr[:N]))

            if max_index != N-1:
                arr = self.flip(arr, max_index + 1)
                arr = self.flip(arr, N)
            N -= 1

        return self.result

    def flip(self, arr, k):
        self.result.append(k)
        arr[:k] = arr[:k][::-1]
        return arr

File: 0969-pancake-sorting/0969-pancake-sorting.cpp
class Solution {
public:
    vector<int> pancakeSort(vector<int>& arr) {
        int n=arr.size();
        vector<int> ans;
        for(int i=n-1; i>0; i--) {
            int maxi=INT_MIN,ind,j=0;
            for(j=0; j<=i; j++) {
                if(maxi < arr[j]) {
                    maxi = arr[j];
                    ind = j;
                }
            }
            if(ind==i)
                continue;
            if(j!=0) {
                ans.push_back(ind+1);
                reverse(arr.begin(),arr.begin()+ind+1);
            }
            ans.push_back(i+1);
            reverse(arr.begin(),arr.begin()+i+1);
        }
        return ans;
    }
};